src/java.base/share/classes/java/util/UUID.java

```diff
@@ -511,6 +511,12 @@ public boolean equals(Object obj) {
      */
     @Override
     public int compareTo(UUID val) {
+        // Compare the timstamps if both UUIDs are timed UUID
+        // Do not do this if the timestamps are the same
+        if (version() == 1 && val.version() == 1 && timestamp() != val.timestamp()) {
+            return Long.compare(timestamp(), val.timestamp());
+        }
+
```

Comment:
Are you sure this compareTo method is transitive? For example: given 2 version 1 based UUIDs, `a` and `b`, and given the 3rd random UUID `c`, lets say `a < b` and `b < c`. Is it true that `a < c` in all such possible `a`, `b` and `c` ?

Reply:
It should be. If a comparison is done between a timed uuid and a regular one, the timestamp comparison will not be done and the original method will be used.